orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py

```diff
@@ -127,7 +127,10 @@ def __init__(self, module, debug_options: DebugOptions, fallback_manager: _Fallb
         # List of opcodes to be considered safe to move before/after cast operation if propagate_cast_ops_level is zero.
         self._propagate_cast_ops_allow = []
         # Whether allow fusion of layer norm subgraph if doing so will cause modified precision.
```

Comment:
can you explain the comment - "will cause modified precision"?

Reply:
I actually don't know who initially wrote the fusion and this flag, and also this comment here. The situation here is if it's mixed precision compute, there is Cast to cast input from fp16 to fp32, the whole graph is computed in fp32, and there is another Cast from fp32 to fp16 for the output, I think the author should be worrying about the precision loss to fuse it (the input and output are all fp16, Cast is fused). But actually if we look into the LayerNorm CUDA kernel impl (which is borrowed from Apex), we can see that the input/output type is T, but the internal compute are actually in type U (all data will be casted to type U for compute), and normally U is higher than T in precision, i.e., if T is fp16, fp32, U is fp32, if T is double, U is also double. So logically this flag is actually not needed, we will not loss precision during the compute. (I didn't check the CPU impl, but this type T and U is defined in schema, so logically all EPs should follow the some precision logic). And this flag currently is for SimplifiedLayerNorm only, we can see for LayerNorm fusion, we will not use such flag, and will fuse the graph directly even it's mixed precision case.

Reply:
so instead of changing the value here - can we simply get rid of this flag?

Reply:
Yes. Just removed it in new commit.